### ISSUE TITLE
applemedia patch breaks coremedia on 10.7+

### ISSUE THREAD

**Issue by wicknix** [rmottola/Arctic-Fox#50]
While the --disable-applemedia switch works to bypass coremedia on 10.6, leaving it enabled on 10.7+ it gets ignored and the browser builds without mp3 support. This makes sites like soundcould.com and shoutcast.com unusable. Please take a look at configure.in and check why coremedia gets ignored when it's not disabled. Thanks.

**Comment by rmottola** [comment]
I found a trivial error in the test of configure! It works for me, in that 10.7 has mp3 support, 10.6 compiles but doesn't. (there is no fallback, I think?) please try

**Comment by rmottola** [comment]
maybe this makes the file configure.in.mp3-10.7 obsolete?

**Comment by wicknix** [comment]
Nice. I'll test it again soon. Yes, the old configure.in.mp3-10.7 is quite obsolete now after all the changes to configure.in and should probably be removed.

**Comment by wicknix** [comment]
Unfortunately that didn't fix it. Still no coremedia found by configure resulting in no mp3 playback.
![shoutcast-no-mp3](https://user-images.githubusercontent.com/39230578/124226644-0f9d8900-dacf-11eb-9f4a-256b0265bc73.png)

![html5test-no-mp3](https://user-images.githubusercontent.com/39230578/124226664-16c49700-dacf-11eb-969b-6940feb46e7c.png)

**Comment by wicknix** [comment]
I just replaced the applemedia lines with the original code and it works on 10.7+ again.
Lines 5064-5066:
if test "$MOZ_WIDGET_TOOLKIT" = "cocoa"; then
    MOZ_APPLEMEDIA=1
fi

**Comment by rmottola** [comment]
Did you use "dev" branch with the new fix? It works fine for me.

**Comment by wicknix** [comment]
Yes dev branch. Didn't work as seen in the posted pics. Works fine with the above changes though.

**Comment by rmottola** [comment]
I found out why it didn't work.... it is a matter of defaults. I attempted to fix it but it breaks non-apple.

I have a more (in)convenient solution. Just add --enable-applemedia explicitely on 10.7 and up, even if it is counter-intuitive. I understand that if disable is set "" (empty) is used as a value, if "not" (= enabled) it should be 1. Thus the later test. It might be some configure trickery, I studied above tests and found some inconsistencies in handling these defaults too
